### PR TITLE
Retract invitations

### PR DESCRIPTION
**Pull request by MohanL**
fixes https://github.com/exercism/exercism.io/issues/3185

This PR adds the functionality to retract invitations.

### CODE REVIEW

File: app/routes/teams.rb [hunk 1]
```diff
@@ -221,6 +221,17 @@ class Teams < Core
         end
       end
 
+      # Delete a team invitation.
+      delete '/teams/:slug/invitations/:username' do |slug, username|
+        please_login
+        only_for_team_managers(slug, "You are not allowed to retract invitations.") do |team|
+          team.dismissInvitation(username)
```
Comment (kytrinyx) [inline app/routes/teams.rb]: method calls in ruby should use `snake_case` rather than `mixedCase`.

**Comment by kytrinyx** [comment]
That was quick! Thank you :heart:

File: lib/exercism/team.rb [hunk 2]
```diff
@@ -88,6 +88,13 @@ def dismiss(username)
     members.delete(user)
   end
 
+  def dismiss_invitation(username)
+    user = User.find_by_username(username)
+    return if user.nil?
+
+    TeamMembershipInvite.where(user: user, team: self).map(&:destroy)
```
Comment (bernardoamc) [inline lib/exercism/team.rb]: Could we replace this `map` by a `destroy_all` call?

Comment (kytrinyx) [inline lib/exercism/team.rb]: That's a good call. @MohanL do you have time to make this change?

**Comment by bernardoamc** [comment]
Do we need a test for this new feature/route? I see value in adding it, but we could do it in a new PR if we don't want to block this one. :)

File: lib/exercism/team.rb (hunk 2 above)
Comment (MohanL) [inline lib/exercism/team.rb]: working on it :) @kytrinyx @bernardoamc

**Comment by MohanL** [comment]
@bernardoamc sure, I can definitely work on that

**Comment by bernardoamc** [comment]
Wow, you are fast! ⚡️

**Comment by c1505** [comment]
Looks good to me :)    

Potentially add an acceptance test in test/acceptance/team_test.rb.  Not sure if you guys are wanting them for new features or not.  

Looking at this I realized that inviting a user to your team will create a user if they don't exist.  Nothing to do with any of the changes here.  Assuming that it isn't desired behavior.  I'll open an issue.

**Comment by MohanL** [comment]
@bernardoamc @kytrinyx  what do you think about @c1505 's idea  ?
If we add another test to cover the notification side of the invitee, would it be better ?

**Comment by kytrinyx** [comment]
> Looking at this I realized that inviting a user to your team will create a user if they don't exist.

I think that's probably the desired behavior. We won't email anyone, of course, but if they then log in, the team invitation should be waiting for them.

We should make sure that flow actually works, though!

> what do you think about @c1505 's idea?

Would you clarify what idea you mean? Is it a test? What would we be testing? What is the risk/what could potentially break that we want to protect against?

**Comment by MohanL** [comment]
@kytrinyx 
it is a test to check if the team notifications received by the invitee is retracted. 
I could probably add `NotificationCount` test in the current test case. What do you think ?

**Comment by MohanL** [comment]
@kytrinyx @bernardoamc 
please have a review to see is there anything more I could do for this PR ? Thank you ! :)

**Comment by bernardoamc** [comment]
Looks pretty solid @MohanL, thanks! <3

**Comment by MohanL** [comment]
I guess this PR is good now ? then I will squash all the commits 😄 Thank you so much @kytrinyx @bernardoamc @c1505

**Comment by kytrinyx** [comment]
@MohanL Yeah, this looks great. Would you squash the commits and rebase onto the most recent master?

**Comment by MohanL** [comment]
No problem ! :smiley

**Comment by kytrinyx** [comment]
> Would you squash the commits and rebase onto the most recent master?

@MohanL When squashing and rebasing be sure to push to the same branch as before in order to update the pull request. If you need help with the workflow, please ask.

**Pull request closed without merging**